Vicare Gas & Power consumption summary sensors

oischinger opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Adds new sensors for Gas & Power consumption available on some Viessmann devices.
Note that these might seem like a duplication of existing sensors like gas_consumption_heating_this_year however these are different API endpoints and it is not completely clear why Viessmann created different API endpoints. Since we do not know which sensors are available for which device we support both in this integration. It seems that either the summary or the already existing sensors are available. Never both at the same time.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

oischinger committed:
Vicare Gas & Power consumption summary sensors

frenck commented on homeassistant/components/vicare/sensor.py:
```diff
@@ -132,6 +133,134 @@ class ViCareSensorEntityDescription(SensorEntityDescription, ViCareRequiredKeysM
+    ),
+    ViCareSensorEntityDescription(
+        key="power_dhw_summary_consumption_heating_currentmonth",
+        name="Power consumption of hot water gas heating current month",
```
All these sensors are using "Power" in their name/description, but are actually Energy sensors (and thus not Power).

oischinger committed:
Rename from power to energy

oischinger committed:
Rename also other sensors from power -> energy

oischinger commented:
I rebased and added 2 commits:

1. The first to fix the sensor keys and names from this commit
2. the 2nd to fix existing sensors that also used "power" instead of "energy". I did not touch the keys though to avoid breaking changes

Styx85 commented:
Can anything be done to get this merged. Actually the Viessman PT2 is relying on this kind of summary sensors, thus the gas reporting for the energy board is incomplete.

frenck commented on homeassistant/components/vicare/sensor.py:
```diff
@@ -132,6 +133,134 @@ class ViCareSensorEntityDescription(SensorEntityDescription, ViCareRequiredKeysM
+        native_unit_of_measurement=VOLUME_CUBIC_METERS,
+        value_getter=lambda api: api.getGasSummaryConsumptionDomesticHotWaterLastSevenDays(),
+        unit_getter=lambda api: api.getGasSummaryConsumptionDomesticHotWaterUnit(),
+        state_class=SensorStateClass.TOTAL_INCREASING,
```
Isn't this sensor a moving window? As in, every day, the last 7 days would shift, right?

oischinger replied:
I cannot answer that since the sensor is not provided by my device.
@Styx85 or @killer0071234 can you answer that?

Does this sensor cover the last 7 days or is it more like the current week?

Styx85 replied:
@oischinger 
Hi for my Device "Vitovalor PT 2" it is the current week (mo-su). I just did a Screenshot in the app to reassure how it is reported.
![Screenshot_2022-04-26-20-40-40-996_com viessmann vicare](https://user-images.githubusercontent.com/32103696/165370004-6acb4a4e-9969-47f2-ad29-e20c23fa4584.jpg)

Styx85 replied:
@oischinger 
As I mentioned here #https://github.com/somm15/PyViCare/issues/246 the Viessmann app seems to use three different api calls / reportings and does the rest as internal calculation. If it helps to clarify what is reported.

oischinger replied:
Can you show a screenhot of the "days"-section?
does it show the last seven days or monday-Today?

Styx85 replied:
It shows the last seven days and additionaly the current day.
![Screenshot_2022-04-26-21-14-55-433_com viessmann vicare](https://user-images.githubusercontent.com/32103696/165375511-d7412e8b-4125-4bca-bfd8-1ffa19a96b9c.jpg)

oischinger replied:
I wonder if this sensor is actually useful in context of home assistant where you would rather use e.g. the consumption/year.
Either we remove it or we remove the `SensorStateClass.TOTAL_INCREASING`

frenck replied:
IMHO, you need only 1 total_increasing for the energy dashboard. It doesn't matter much which one.

oischinger replied:
I wonder if this sensor is actually useful in context of home assistant where you would rather use e.g. the consumption/year.
Either we remove it or we remove the `SensorStateClass.TOTAL_INCREASING`

Styx85 replied:
Correct, some like to reproduce the app display in ha as well.
Anyhow. With these changes the Vitovalor still is missing the total gas consumption. Or am i missing something?

oischinger replied:
I removed the "last seven days" sensor since it's the most problematic one concerning `SensorStateClass.TOTAL_INCREASING`

The others might be useful to someone

frenck commented on homeassistant/components/vicare/sensor.py:
```diff
@@ -142,31 +271,31 @@ class ViCareSensorEntityDescription(SensorEntityDescription, ViCareRequiredKeysM
     ViCareSensorEntityDescription(
         key="power_production_this_week",
-        name="Power production this week",
+        name="PoEnergyEnergywer production this week",
```
Something went wrong here 😄

oischinger replied:
oh. thanks! 😊

oischinger committed:
Fix typo

oischinger commented:
I rebased and fixed the typo

oischinger committed:
Remove last seven days sensor

frenck reviewed (approved):
Thanks, @oischinger 👍
